src/server/jobs/collection-contest-vimeo-upload.ts

```diff
@@ -0,0 +1,198 @@
+          JOIN "Image" i ON i.id = ci."imageId"
+          JOIN "Post" p ON p.id = i."postId"
+          JOIN "User" u ON u.id = p."userId"
+          WHERE ci."collectionId" = ${collection.id}
+            AND ci."status" = 'ACCEPTED'
+            AND i.type = 'video'
+            AND i."ingestion" = 'Scanned'
+            -- We only want to upload videos that are longer than 30 seconds
+            AND (i.metadata->'duration')::int > 30  
+            AND (i.metadata->'vimeoVideoId') IS NULL
+            AND ci."updatedAt" > ${lastRun}
+          -- Ensures that we try to upload smaller videos first as a safeguard.
+          ORDER BY i.metadata->'size' ASC
+        `;
+
+        for (const item of collectionItems) {
+          try {
+            if (item.metadata.vimeoVideoId) {
+              console.log(`Video already uploaded ${item.imageId}`);
+              continue;
+            }
+
+            if ((item.metadata.vimeoUploadAttempt ?? 0) > ATTEMPT_LIMIT) {
+              console.log(`Video upload attempts exceeded ${item.imageId}`);
+              continue;
+            }
+
+            if (!!item.metadata.vimeoUploadEnqueuedAt) {
+              continue;
+            }
+
+            if (env.VIMEO_VIDEO_UPLOAD_URL) {
+              const callbackUrl = `${env.NEXT_PUBLIC_BASE_URL}/api/webhooks/vimeo-upload?token=${env.WEBHOOK_TOKEN}`;
+
+              const res = await fetch(env.VIMEO_VIDEO_UPLOAD_URL, {
+                method: 'POST',
+                headers: {
+                  'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({
+                  imageId: item.imageId,
+                  sourceUrl: getEdgeUrl(item.imageUrl, {
+                    original: true,
+                  }),
+                  callbackUrl,
+                }),
+              });
+
+              if (!res.ok) {
+                throw new Error(`Failed to upload video ${item.imageId}`);
+              }
+
+              await dbWrite.image.update({
+                where: { id: item.imageId },
+                data: {
+                  metadata: {
+                    ...item.metadata,
+                    vimeoUploadEnqueuedAt: new Date().toISOString(),
+                  },
+                },
+              });
+            } else {
+              // Assume we're uploading from the CRON Job:
+              const userProfile = `${env.NEXT_PUBLIC_BASE_URL}/user/${item.username}`;
+
+              const data: { link: string } = await uploadVimeoVideo({
+                url: item.imageUrl,
+                accessToken: env.VIMEO_ACCESS_TOKEN,
+                title: item.title,
+                description: `
+                  ${sanitize(item.detail, {
+                    allowedTags: [],
+                    allowedAttributes: {},
+                  })}
+
+                  Created by ${item.username}:
+                  ${userProfile}
+
+                  Check out more entries at:
```

Comment:
Probably a good idea to mention civitai somewhere
```suggestion
                  Check out more entries at Civitai Project Odyssey:
```